pandas/tseries/resample.py

```diff
@@ -764,6 +768,20 @@ class TimedeltaResampler(DatetimeIndexResampler):
     def _get_binner_for_time(self):
         return self.groupby._get_time_delta_bins(self.ax)
 
+    def _adjust_binner_for_upsample(self, binner):
+        """ adjust our binner when upsampling """
+        ax = self.ax
+
+        if is_subperiod(ax.freq, self.freq):
```

Comment:
doesn't work. I think there is something going on w.r.t. how the values come back for DTI vs. TDI, meaning TDI always adds a stamp on (then removes it here). There is lots of internal logic, e.g. ``_adjust_bin_edges`` that is doing things.